include/aws/crt/ByteBuf.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/crt/Outcome.h>
+#include <aws/crt/Types.h>
+
+#include <aws/common/byte_buf.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+        class ByteCursor
+        {
+          public:
+            ByteCursor() noexcept;
+            ByteCursor(const ByteCursor &cursor) noexcept;
+            explicit ByteCursor(const char *str) noexcept;
+            explicit ByteCursor(const String &str) noexcept;
+            explicit ByteCursor(aws_byte_cursor cursor) noexcept;
+            explicit ByteCursor(aws_byte_cursor *cursor) noexcept;
```

Comment:
I would be more explicit about what's going on with the by-pointer constructor. It has *radically* different behavior than the rest (modifying the C++ object, also modifies the thing used to construct it). Maybe:

static ByteCursor InitWrapper(aws_byte_cursor *cursorToWrap);